docs/framework/winforms/controls/how-to-bind-data-to-the-windows-forms-datagridview-control.md

```diff
@@ -1,57 +1,49 @@
+
+Visual Studio has extensive support for data binding to the DataGridView control. For more information, see [How to: Bind data to the Windows Forms DataGridView control using the Designer](https://msdn.microsoft.com/library/33w255ac\(v=vs.110\)).  
+
+To connect a DataGridView control to data:
+
+1. Implement a method to handle the details of retrieving the data. The following code example implements a `GetData` method that initializes a <xref:System.Data.SqlClient.SqlDataAdapter>, and uses it to populate a <xref:System.Data.DataTable>. It then binds the <xref:System.Data.DataTable> to the <xref:System.Windows.Forms.BindingSource>. 
+
+2. In the form's <xref:System.Windows.Forms.Form.Load> event handler, bind the <xref:System.Windows.Forms.DataGridView> control to the <xref:System.Windows.Forms.BindingSource>, and call the `GetData` method to retrieve the data.  
+
+## Example
+
+This complete code example retrieves data from a database to populate a DataGridView control in a Windows form. The form also has buttons to reload data and submit changes to the database.  
+
+This example requires: 
+
+- Access to a Northwind SQL Server sample database. For more information about installing the Northwind sample database, see [Get the sample databases for ADO.NET code samples](../../data/adonet/sql/linq/downloading-sample-databases.md). 
+
+- References to the System, System.Windows.Forms, System.Data, and System.Xml assemblies.  
+
+To build and run this example, paste the code into the *Form1* code file in a new Windows Forms project.  For more information, see [How to: Compile and run a complete Windows Forms code example using Visual Studio](https://msdn.microsoft.com/library/Bb129228\(v=vs.110\)). For information about building from the C# or Visual Basic command line, see [Build from the command line](/visual-basic/reference/command-line-compiler/building-from-the-command-line) and [Command-line building with csc.exe](/csharp/language-reference/compiler-options/command-line-building-with-csc-exe).  
+  
+Populate the `connectionString` variable in the example with the values for your Northwind SQL Server sample database connection. Windows Authentication, also called integrated security, is a more secure way to connect to the database than storing a password in the connection string. For more information about connection security, see [Protect connection information](../../data/adonet/protecting-connection-information.md).  
+
```

Comment:
For both source code files:
- You can remove the snippet tags.
- In the two code references here, you can delete `#00` so that the code reference is simply to the entire file and not to an embedded snippet. That's marginally more performant, since the build doesn't need to actually parse the file.